pallets/rc-migrator/src/types.rs

```diff
@@ -212,6 +213,45 @@ impl<Status: MigrationStatus, Default: Get<Weight>> Get<Weight> for ZeroWeightOr
 		Status::is_ongoing().then(Weight::zero).unwrap_or_else(Default::get)
 	}
 }
+
+/// A wrapper around `Inner` that routes messages through `Inner` unless `Exception` is true and
+/// `MigrationState` is ongoing.
+pub struct RouteInnerWithException<Inner, Exception, MigrationState>(
+	PhantomData<(Inner, Exception, MigrationState)>,
+);
```

Comment:
some improvement can be made to `Exception` naming to explain that it is a condition applied to the `destination`, but just nice to have, the type defined here will be short-lived anyway (removed after AHM)

Reply:
> some improvement can be made to `Exception` naming to explain that it is a condition applied to the `destination`

Agree, I would vote for renaming `Exception` to `BlockFor` -  `BlockFor::contains(destination`

Reply:
ok, no problem, we can keep as it is.


```
/// A wrapper around `Inner` that routes messages through `Inner` unless `Exception` is true and
/// `MigrationState` is ongoing.
pub struct RouteInnerUnlessException<Inner, Exception, MigrationState>(

or shorther

pub struct RouteInnerUnless<Inner, Exception, MigrationState>(

or even shorter

pub struct RouteUnless<Inner, Exception, MigrationState>(
```

Reply:
I asked a large language model 
![Screenshot 2025-07-03 at 10 26 47](https://github.com/user-attachments/assets/fa98bf17-6c57-4200-bff0-87f355dc881f)

Reply:
haha, suggestions above were also based on AI :D :D :D,
let's keep it as it is :)